experimental/devices/as7262/as7262.go

```diff
@@ -0,0 +1,532 @@
+// Copyright 2018 The Periph Authors. All rights reserved.
+// Use of this source code is governed under the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+// +build go1.7
+
+package as7262
+
+import (
+	"context"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"math"
+	"sync"
+	"time"
+
+	"periph.io/x/periph/conn"
+	"periph.io/x/periph/conn/gpio"
+	"periph.io/x/periph/conn/i2c"
+	"periph.io/x/periph/conn/physic"
+)
+
+// Opts holds the configuration options.
+type Opts struct {
+	InterruptPin gpio.PinIn
+	Gain         Gain
+}
+
+// DefaultOpts are the recommended default options.
+var DefaultOpts = Opts{
+	InterruptPin: nil,
+	Gain:         G1x,
+}
+
+// New opens a handle to an AS7262 sensor.
+func New(bus i2c.Bus, opts *Opts) (*Dev, error) {
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
periph doesn't use context yet but that's an interesting question.

Reply:
I'd forgot 1.5 didn't have context. It can be rewritten with context and maintain same API.